src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -2014,7 +2014,20 @@ fn disambiguator_error(
     link_range: Range<usize>,
     msg: &str,
 ) {
-    report_diagnostic(cx.tcx, BROKEN_INTRA_DOC_LINKS, msg, item, dox, &link_range, |_diag, _sp| {});
+    report_diagnostic(cx.tcx, BROKEN_INTRA_DOC_LINKS, msg, item, dox, &link_range, |diag, _sp| {
+        let channel = match env!("CFG_RELEASE_CHANNEL") {
+            "stable" => env!("CFG_RELEASE_NUM"),
+            "beta" => "beta",
+            "nightly" | "dev" => "nightly",
+            // custom build of rustdoc maybe? link to the stable docs just in case
+            _ => "",
+        };
+        let msg = format!(
+            "see https://doc.rust-lang.org/{}/rustdoc/linking-to-items-by-name.html#namespaces-and-disambiguators for more info about disambiguators",
```

Comment:
nit: Could you break this line into two? Perhaps right after the URL? To be clear, this is just to make the code easier to read; I'm not suggesting an output change.